docs/proposals/PROPOSAL_H.md

```diff
@@ -0,0 +1,281 @@
+      "cpu-cpuid.AVX512FP16": "true",
+      "kernel-selinux.enabled": "false",
+      "kernel-config.PREEMPT": "true",
+      "kernel-version.full": "[4.19, 5.16)",
+      "system.glibc": "[2.31, 2.37]",
+      "pci-15B3-020D.present": "true",
+    },
+    {
+      "cpu-model.vendor_id": "AuthenticAMD",
+      "cpu-cpuid.FPHP": "true",
+      "kernel-selinux.enabled": "false",
+      "kernel-config.PREEMPT": "true",
+      "kernel-version.full": "[4.19, 5.16)",
+      "system.glibc": "[2.31, 2.37]",
+      "pci-1002-67ff.present": "true",
+      "annotation": "only work with AMD CPU and AMD GPU"
+    }
+  ]
+}
+```
+
+### Example - image is compatible with optimal hardware
+
+```json
+{
+  "compatibilities": [
+    {
+      "cpu-model.vendor_id": "GenuineIntel",
+      "cpu-cpuid.AMXFP16": "true",
+      "kernel-selinux.enabled": "false",
+      "kernel-config.PREEMPT": "true",
+      "kernel-version.full": "[4.19, 5.16)",
+      "system.glibc": "[2.31, 2.37]",
+      "system.openmpi": "[4.1.6, 4.1.6]",
+      "pci-15B3-019D.present": "true",
+      "compatibility-score": 50,
+    },
+    {
+      "cpu-model.vendor_id": "GenuineIntel",
+      "cpu-cpuid.AVX512FP16": "true",
+      "kernel-selinux.enabled": "false",
+      "kernel-config.PREEMPT": "true",
+      "kernel-version.full": "[4.19, 5.16)",
+      "system.glibc": "[2.31, 2.37]",
+      "system.openmpi": "[4.1.6, 4.1.6]",
+      "pci-15B3-020D.present": "true",
+      "compatibility-score": 100,
+    },
+    {
+      "cpu-model.vendor_id": "AuthenticAMD",
+      "cpu-cpuid.FPHP": "true",
+      "kernel-selinux.enabled": "false",
+      "kernel-config.PREEMPT": "true",
+      "kernel-version.full": "[4.19, 5.16)",
+      "system.glibc": "[2.31, 2.37]",
+      "system.openmpi": "[4.1.6, 4.1.6]",
+      "pci-1002-67ff.present": "true",
+      "compatibility-score": 80,
+      "annotation": "only work with AMD CPU and AMD GPU"
+    }
+  ]
+}
+```
+
+## Alternative place to hold compatibility set
+
+Considering the compatibilities are composed of a list of compatibility set, just similar as the idea in proposal E, the platform.features could be reused.
+
+Each compatibility set will be encoded into a string, then the platform.features can hold all the compatibility artifact content, the downside is the platform.features content is hard to read for human.
+
+Then it's the image author's(or image provider's) freedom to use standalone artifact or platform.features to express the compatibility requirements, the format will be kept same.
+
+To reduce the load to registry, if compatibilities are stored in platform.features(a new filed inside comaptibility set can be used to identify this case), no further api request should be sent to registry to discover the comaptibility artifact.
+
+## Compatibility and NFD filed labels
+
+If the field is required in image compatibility artifact, but does not present in NFD labels, the new feature lable should be discussed and contributed into NFD.
```

Comment:
If I understand this correctly, it means that the proposal suggest to use only NFD specific labels, correct?

Reply:
It is also possible to provide custom labels under NFD, without contributing to the project itself.

Reply:
Mentioned that NFD labels can not meet all OCI compatibility demands,  and some times NFD labels have different purpose. It's better to define OCI labels, for example, in NFD, kernel-config.option-1: "true", kernel-config.option-2: "true", kernel-config.option-n: "true", NFD labels are dynamicly created according to features. 
It's reasonable in NFD to use dynamic labels, beacuse these labels will be applied to nodes. OCI label can be defined in a simpler way, it could be oci.kernel.config: "option-1, option-2, option-n", fixed label name is more friendly, and can be translated to NFD labels. Collaboration is good.

Reply:
So I understand the idea is to go back to the original plan introduced in the proposal B, to define well known labels with the following categories:
* CPU
* Kernel
* Memory
* Network
* PCI
* USB
* Storage
* System
* Local (custom labels)

Reply:
I expressed the idea of know labels in the proposal E that could be well combined with this proposal.
https://github.com/opencontainers/wg-image-compatibility/pull/13#issuecomment-2040131560